brave/brave-core
# Incorrect block explorer launched with the correct transaction hash

Implemented an ability to provide chain_id for each TransactionInfo returned to front end and saving the same to preferences.
For example before this change the transaction information in the preference looks like:
![image](https://user-images.githubusercontent.com/118171981/221305278-51830f4f-6c22-4283-8a95-6bcde3d60277.png)
But after changes applying and migration executing:
![image](https://user-images.githubusercontent.com/118171981/221298481-81e77cb7-f73d-46c6-a6f9-fa74d8599113.png)


Resolves https://github.com/brave/brave-browser/issues/28754


## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues/28754) for my issue
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1. Make sure that you have preferences data which is not migrated (i.e. transaction info looks like on the first image from the PR description)
2. Install/update browser, then start it.
3. Open preferences file and make sure that migration is applied, and each transaction info contains new "chain_id" property with valid value
4. Send any amount of ETH, SOL and FIL coin and make sure that in the preferences file each new transaction info saved with new "chain_id" property
